src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,409 @@
+    displacement, min_length, structure, sym_reduce, symprec, conventional
+):
+    transformation = CubicSupercellTransformation(min_length=min_length)
+    transformation.apply_transformation(structure=structure)
+    supercell_matrix = transformation.transformation_matrix.tolist()
+
+    cell = get_phonopy_structure(structure)
+    if not conventional:
+        phonon = Phonopy(
+            cell,
+            supercell_matrix,
+            primitive_matrix=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]],
+            factor=VaspToTHz,
+            symprec=symprec,
+            is_symmetry=sym_reduce,
+        )
+    else:
+        spa = SpacegroupAnalyzer(structure)
+        matrix = spa.get_conventional_to_primitive_transformation_matrix()
+
+        phonon = Phonopy(
+            cell,
+            supercell_matrix,
+            primitive_matrix=matrix,
+            factor=VaspToTHz,
+            symprec=symprec,
+            is_symmetry=sym_reduce,
+        )
+
+    phonon.generate_displacements(distance=displacement)
+    return phonon
+
+
+# check if this can also be replaced with something better
+def get_kpath(structure: Structure, **kpath_kwargs):
+    """
+    get high-symmetry points in k-space
+    Args:
+        structure: Structure Object
+    Returns:
+    """
+    highsymmkpath = HighSymmKpath(structure, **kpath_kwargs)
+    kpath = highsymmkpath.kpath
+    path = copy.deepcopy(kpath["path"])
+
+    for ilabelset, labelset in enumerate(kpath["path"]):
+        for ilabel, label in enumerate(labelset):
+            path[ilabelset][ilabel] = kpath["kpoints"][label]
+    return kpath["kpoints"], path
+
+
+# TODO: check all parameters again
+@job
+def generate_phonon_displacements(
+    structure: Structure,
+    displacement: float = 0.01,
+    min_length: float = 4.0,
+    conventional: bool = False,
+    symprec: float = SETTINGS.SYMPREC,
+    sym_reduce: bool = True,
+):
+    """
+    Generate elastic deformations.
+
+    Parameters
+    ----------
+    structure : Structure
+        A pymatgen structure object.
+    displacement : float
+        The displacement to be applied to the structure.
+    min_length  : float
+        minimum supercell size.
+    symprec : float
+        The symprec to use for the spacegroup analyzer.
+    sym_reduce : bool
+        Whether to reduce the symmetry of the structure.
+
+    Returns
+    -------
+    List[Deformation]
+        A list of diplacements.
+    """
+    # TODO: use functions from pymatgen instead?
+    phonon = get_phonon_object(
+        displacement=displacement,
+        min_length=min_length,
+        structure=structure,
+        sym_reduce=sym_reduce,
+        symprec=symprec,
+        conventional=conventional,
+    )
+    supercells = phonon.supercells_with_displacements
+
+    displacements = []
+    for cell in supercells:
+        displacements.append(get_pmg_structure(cell))
+    return displacements
+
+
+@job(output_schema=PhononBSDOSDoc)
```

Comment:
I think phonon band structures and density of states should be stored in the data store (for large objects). You can do this automatically using:

```suggestion
from pymatgen.phonon.bandstructure import PhononBandStructure
from pymatgen.phonon.dos import PhononDos

@job(output_schema=PhononBSDOSDoc, data=[PhononDos, PhononBandStructure])
```